test/test_formatting.py

```diff
@@ -24,6 +24,9 @@ def test_color():
     assert color(text) == text
     assert color(text, colors.PINK) == '\x0313' + text + '\x03'
     assert color(text, colors.PINK, colors.TEAL) == '\x0313,10' + text + '\x03'
+    assert color(text, '3') == '\x0303' + text + '\x03'  # ensure single-digit str is zero-padded
+    assert color(text, 3) == '\x0303' + text + '\x03'  # ensure ints are zero-padded
+    assert color(text, 0) == '\x0300' + text + '\x03'  # ensure that color 0 isn't treated as boolean false
     pytest.raises(ValueError, color, text, 100)
     pytest.raises(ValueError, color, text, 'INVALID')
```

Comment:
One more group of tests is required here: check what happen with an empty string.

```suggestion
    pytest.raises(ValueError, color, text, 100)
    pytest.raises(ValueError, color, text, 'INVALID')
    pytest.raises(ValueError, color, text, '')
    # test background color as well
    pytest.raises(ValueError, color, text, '', '')
    pytest.raises(ValueError, color, text, None, '')
    pytest.raises(ValueError, color, text, colors.PINK, '')
    pytest.raises(ValueError, color, text, '', colors.PINK)
```

Reply:
Good point.

Relatedly, I thought about adding type hints to this stuff as long as I was touching it, but couldn't find a sensible way to indicate "this takes an int _in a certain value range_, or a str _but not an empty one_". If there is none and we're meant to just annotate the type and let `ValueError` etc. handle the acceptable values within that type, then sure, type hinting this module could be a future PR of its own.